src/shipit_pulse_listener/shipit_pulse_listener/listener.py

```diff
@@ -1,120 +1,100 @@
-from taskcluster.utils import slugId
-from cli_common.pulse import create_consumer, run_consumer
+from cli_common.pulse import run_consumer
 from cli_common.taskcluster import TaskclusterClient
 from cli_common.log import get_logger
-import copy
-import json
-import re
+from shipit_pulse_listener.hook import Hook
+import asyncio
 
 logger = get_logger(__name__)
 
-HOOK_REGEX = re.compile(r'([\w\-_]+):([\w\-_]+)')
 
-
-class PulseListener(object):
+class HookStaticAnalysis(Hook):
     """
-    Listen to pulse messages and trigger new tasks
+    Taskcluster hook handling the static analysis
     """
-    def __init__(self, secrets_path, client_id=None, access_token=None):
-        self.taskcluster = TaskclusterClient(client_id, access_token)
-        self.tasks = []
-
-        # Fetch pulse credentials from TC secrets
-        logger.info('Using secrets', path=secrets_path)
-        required = ('PULSE_USER', 'PULSE_PASSWORD', 'PULSE_QUEUE')
-        secrets = self.taskcluster.get_secrets(secrets_path, required)
-
-        # Use pulse consumer from bot_common
-        self.consumer = create_consumer(
-            secrets['PULSE_USER'],
-            secrets['PULSE_PASSWORD'],
-            secrets['PULSE_QUEUE'],
-            secrets.get('PULSE_TOPIC', '#'),
-            self.got_message
+    def __init__(self, branch):
+        super().__init__(
+            'project-releng',
+            'services-{}-shipit-static-analysis-bot'.format(branch),
+            'exchange/mozreview/',
+            'mozreview.review.published',
         )
 
-    def load_hooks(self, hooks):
+    def parse_payload(self, payload):
         """
-        Load task payloads from hooks definitiions
+        Start new tasks for every bugzilla id
         """
-        assert len(hooks) > 0, \
-            'Missing hooks definitions'
-
-        service = self.taskcluster.get_hooks_service()
-        for hook in hooks:
-
-            # Get hook definitions
-            result = HOOK_REGEX.search(hook)
-            if result is None:
-                logger.warn('Invalid hook definition', definition=hook)
-                continue
-
-            hookGroupId, hookId = result.groups()
-            logger.info('Using hook definition', group=hookGroupId, id=hookId)
-
-            # Get task payload from hook
-            hook_payload = service.hook(hookGroupId, hookId)
-            self.tasks.append(hook_payload['task'])
-
-        if not self.tasks:
-            raise Exception('No tasks to run.')
+        # Filter on repo url
+        landing_repository_url = payload.get('landing_repository_url')
+        if not landing_repository_url:
+            raise Exception('Missing landinf repository url in payload')
+        if 'reviewboard-hg.mozilla.org' not in landing_repository_url:
+            logger.warn('Skipping this message, invalid landing repository url', url=landing_repository_url)  # noqa
+            return
+
+        # Extract commits
+        bugzilla_id = payload.get('id')
```

Comment:
Nit: I don't see any `id` payload key in the Pulse message's [documentation](https://mozilla-version-control-tools.readthedocs.io/en/latest/mozreview/notifications.html#mozreview-commits-published) or on [Pulse inspector](https://tools.taskcluster.net/pulse-inspector/#!((exchange:exchange/mozreview/,routingKeyPattern:%23))). The commits appears to have their own `commits` key which is an array (the commit series to be reviewed, and applied by us for static analysis).

Edit: I see that you process the `commits` payload key below.